### PR TITLE
Complex fetcher

### PR DESCRIPTION
**Pull request by barbarahui**

### CODE REVIEW

File: metadata-fetcher/NuxeoFetcher.py [hunk 1]
```diff
@@ -1,40 +1,120 @@
...
+                "X-NXRepository": "default",
+                "X-Authentication-Token": TOKEN
+                }
 
 class NuxeoFetcher(Fetcher):
     def __init__(self, params):
         super(NuxeoFetcher, self).__init__(params)
         self.nuxeo = params.get('nuxeo')
-        if not self.nuxeo.get('current_page_index'):
+        if self.nuxeo.get('current_page_index') is None:
```
[the remark this answers is not in the document]
Comment (barbarahui) [reply metadata-fetcher/NuxeoFetcher.py:38]: Oh yeah that would be SO much better!! Will implement.

File: metadata-fetcher/lambda_function.py [hunk 2]
```diff
@@ -31,8 +31,8 @@ def lambda_handler(payload, context):
     if DEBUG:
         payload = json.loads(payload)
 
-    harvest_type = payload.get('harvest_type')
-    fetcher = get_fetcher(harvest_type)
+    #harvest_type = payload.get('harvest_type')
+    fetcher = NuxeoFetcher(payload)
```
[the remark this answers is not in the document]
Comment (barbarahui) [reply metadata-fetcher/lambda_function.py:35]: Oh hmmm yeah I remember now that I wasn't getting this to work for some reason and so I commented it out, but that was a long time ago... Will look into it.

File: metadata-fetcher/NuxeoFetcher.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (barbarahui) [reply metadata-fetcher/NuxeoFetcher.py]: I ended up changing things quite a bit to hopefully clarify what's going on for the reader. Let me know if this is easier to make sense of!

File: metadata-fetcher/NuxeoFetcher.py [hunk 3]
```diff
@@ -1,40 +1,120 @@
...
+        if self.nuxeo.get('fetch_components') is None:
+            self.nuxeo['fetch_components'] = True
+        if self.nuxeo.get('parent_list') is None:
+            self.nuxeo['parent_list'] = []
+        if self.nuxeo.get('component_count') is None:
+            self.nuxeo['component_count'] = 0
+        if self.nuxeo.get('last_request_doc_count') is None:
+            self.nuxeo['last_request_doc_count'] = 0
+
+    def get_nuxeo_id(self, path):
```
[the remark this answers is not in the document]
Comment (barbarahui) [reply metadata-fetcher/NuxeoFetcher.py]: Again, hopefully things are easier to make sense of now?

File: metadata-fetcher/lambda_function.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (barbarahui) [reply metadata-fetcher/lambda_function.py]: I must've had an old copy of this file in my branch or something? I think it's working the way you intended now.

**Review by amywieliczka — APPROVED** [review]

**Pull request merged**